query-connector/src/app/(pages)/userManagement/utils.ts

```diff
@@ -1,10 +1,101 @@
 import { useSession } from "next-auth/react";
+import { useContext } from "react";
+import { DataContext } from "@/app/shared/DataProvider";
+import { isAuthDisabledClientCheck } from "@/app/utils/auth";
+import { UserRole } from "@/app/models/entities/users";
 
 /**
  * Function that retrieves the role value from the current session object
  * @returns - The UserRole for the current logged-in user
  */
-export function getSessionRole() {
+export function getContextRole() {
```

Comment:
nit: could we just name this `getRole` and leave the implementation details as to "where" out of the function name? Don't think it really matters to the caller how the role gets populated